tags-config-service-impl/src/main/java/org/hypertrace/tag/config/service/TagConfigServiceImpl.java

```diff
@@ -23,9 +30,35 @@
 @Slf4j
 public class TagConfigServiceImpl extends TagConfigServiceGrpc.TagConfigServiceImplBase {
   private final ConfigServiceCoordinator configServiceCoordinator;
+  private final String TAG_CONFIG_SERVICE_CONFIG = "tag.config.service";
+  private final String SYSTEM_TAGS = "system.tags";
+  private final List<Tag> systemTags;
+  private final List<String> systemTagIds;
+  private final List<String> systemTagKeys;
 
-  public TagConfigServiceImpl(Channel configChannel) {
+  public TagConfigServiceImpl(Channel configChannel, Config config) {
     configServiceCoordinator = new ConfigServiceCoordinatorImpl(configChannel);
+    Config tagConfig = config.getConfig(TAG_CONFIG_SERVICE_CONFIG);
+    List<? extends ConfigObject> systemTagsObjectList = tagConfig.getObjectList(SYSTEM_TAGS);
+    systemTags = buildSystemTagList(systemTagsObjectList);
+    systemTagIds =
+        systemTags.stream().map(systemTag -> systemTag.getId()).collect(Collectors.toList());
+    systemTagKeys =
+        systemTags.stream().map(systemTag -> systemTag.getKey()).collect(Collectors.toList());
+  }
+
+  private static List<Tag> buildSystemTagList(List<? extends ConfigObject> configObjectList) {
+    return configObjectList.stream()
+        .map(TagConfigServiceImpl::buildTagFromConfig)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  @SneakyThrows
+  private static Tag buildTagFromConfig(ConfigObject configObject) {
```

Comment:
remove static

Reply:
static is necessary here. IDE complains.